textalyze.js

```diff
@@ -5,20 +5,81 @@
 // comments are here to help you, but please delete them as you go along.
 
 // Open lib/itemCounts.js to get started on the first iteration.
-let itemCounts = require('./lib/itemCounts');
+// let itemCounts = require('./lib/itemCounts');
+
 let frequencyStats = require('./lib/frequencyStats');
 let readFileSync = require('./lib/readFileSync');
 let sanitize = require('./lib/sanitize');
 let stringToChars = require('./lib/stringToChars');
+let stringToWords = require('./lib/stringToWords');
+let wordAnalysis = require('./lib/wordAnalysis');
 let process = require('process');
-let file = process.argv[2];
+let fs = require('fs');
+let helpers = require('./printHelpers');
+
+// Define functions for handling error cases
+function printUsage() {
+  console.log('Usage:');
+  console.log(' node textalyze.js <fileName> <characters OR words>');
+}
+function printError(errorMsg) {
+  console.log(`Error: ${errorMsg}`);
+}
+function printErrorAndExit(errorMsg) {
+  printError(errorMsg);
+  printUsage();
+  process.exit();
+}
+function textalyze() {
+  let file = process.argv[2];
+  let option = process.argv[3];
+
+  // Implement error-handling functions
+  if (file === undefined) {
+    printErrorAndExit('No file given to analyze.');
+  }
+  if (!fs.existsSync(file)) {
+    printErrorAndExit(`No such file ${file}`);
+  }
+
+  if (option === undefined) {
+    printErrorAndExit('No analysis option given.');
+  }
+
+  if (!['words','characters'].includes(option)) {
+    printErrorAndExit('Invalid analysis option.');
+  }
 
-function textalyze(file) {
-  return frequencyStats(stringToChars(sanitize(readFileSync(file))));
+  // Return resulting statistics
+  if (option === 'words') {
+    let data = wordAnalysis.wordStats(stringToWords(sanitize(readFileSync(file))));
+    let wordsData = wordAnalysis.wordSummary(stringToWords(sanitize(readFileSync(file))));
+    console.log();
+    console.log(`This book has ${wordsData[0]} words total.`);
+    console.log(`It used ${wordsData[1]} DIFFERENT words.`);
+    console.log();
+    for (let key of Object.keys(data)) {
+      helpers.print(`The word '${key}' was used ${data[key]} times.`);
+      helpers.printNewLine();
+    }
+    console.log();
+  }
+  if (option === 'characters') {
+    let data = frequencyStats(stringToChars(sanitize(readFileSync(file))));
+    console.log();
+    console.log(' Presenting character analysis of this book:\n');
```

Comment:
Nice!

You _probably_ don't want to use a fixed multiple of the raw percents here. You should figure out how long you want the largest frequency to be and then scale everything relative to that.